client/state/document-head/reducer.js

```diff
@@ -18,7 +18,7 @@ import {
 export function title( state = '', action ) {
 	switch ( action.type ) {
 		case DOCUMENT_HEAD_TITLE_SET:
-			return action.title;
+			return action.title || state;
```

Comment:
> I'd rather we avoid dispatching altogether if we don't intend for the title to be used.

Yeah. The issue here is that returning `undefined` from a reducer yields an error:

> Reducer "title" returned undefined handling "DOCUMENT_HEAD_TITLE_SET"